hotham-openxr-client/src/lib.rs

```diff
@@ -0,0 +1,158 @@
+        *function = transmute::<pfn::WaitSwapchainImage, _>(wait_swapchain_image);
+    } else if name == b"xrSyncActions" {
+        *function = transmute::<pfn::SyncActions, _>(sync_actions);
+    } else if name == b"xrLocateSpace" {
+        *function = transmute::<pfn::LocateSpace, _>(locate_space);
+    } else if name == b"xrGetActionStatePose" {
+        *function = transmute::<pfn::GetActionStatePose, _>(get_action_state_pose);
+    } else if name == b"xrLocateViews" {
+        *function = transmute::<pfn::LocateViews, _>(locate_views);
+    } else if name == b"xrReleaseSwapchainImage" {
+        *function = transmute::<pfn::ReleaseSwapchainImage, _>(release_swapchain_image);
+    } else if name == b"xrEndFrame" {
+        *function = transmute::<pfn::EndFrame, _>(end_frame);
+    } else if name == b"xrRequestExitSession" {
+        *function = transmute::<pfn::RequestExitSession, _>(request_exit_session);
+    } else if name == b"xrDestroySpace" {
+        *function = transmute::<pfn::DestroySpace, _>(destroy_space);
+    } else if name == b"xrDestroyAction" {
+        *function = transmute::<pfn::DestroyAction, _>(destroy_action);
+    } else if name == b"xrDestroyActionSet" {
+        *function = transmute::<pfn::DestroyActionSet, _>(destroy_action_set);
+    } else if name == b"xrDestroySwapchain" {
+        *function = transmute::<pfn::DestroySwapchain, _>(destroy_swapchain);
+    } else if name == b"xrDestroySession" {
+        *function = transmute::<pfn::DestroySession, _>(destroy_session);
+    } else if name == b"xrDestroyInstance" {
+        *function = transmute::<pfn::DestroyInstance, _>(destroy_instance);
+    } else if name == b"xrEnumerateViewConfigurations" {
+        *function = transmute::<pfn::EnumerateViewConfigurations, _>(enumerate_view_configurations);
+    } else if name == b"xrEnumerateReferenceSpaces" {
+        *function = transmute::<pfn::EnumerateReferenceSpaces, _>(enumerate_reference_spaces);
+    } else if name == b"xrGetSystemProperties" {
+        *function = transmute::<pfn::GetSystemProperties, _>(get_system_properties);
+    } else if name == b"xrEnumerateSwapchainFormats" {
+        *function = transmute::<pfn::EnumerateSwapchainFormats, _>(enumerate_swapchain_formats);
+    } else if name == b"xrGetActionStateFloat" {
+        *function = transmute::<pfn::GetActionStateFloat, _>(get_action_state_float);
+    } else if name == b"xrGetActionStateBoolean" {
+        *function = transmute::<pfn::GetActionStateBoolean, _>(get_action_state_boolean);
+    } else if name == b"xrApplyHapticFeedback" {
+        *function = transmute::<pfn::ApplyHapticFeedback, _>(apply_haptic_feedback);
+    } else if name == b"xrEndSession" {
+        *function = transmute::<pfn::EndSession, _>(end_session);
+    } else {
+        let _name = String::from_utf8_unchecked(name.to_vec());
+        unsafe extern "system" fn bang() -> Result {
+            panic!("UNIMPLEMENTED FUNCTION!");
```

Comment:
Could potentially add the name to the panic to make it easier to track down.

Reply:
This is a fun one! So I've tried to pass this in but for whatever reason - potentially FFI weirdness, I am clearly not clever enough to understand those machinations - it isn't possible to return a function that will panic with the correct function name.